python/pyarrow/tests/test_table.py

```diff
@@ -1746,3 +1746,56 @@ def test_table_select():
     result = table.select(['f2'])
     expected = pa.table([a2], ['f2'])
     assert result.equals(expected)
+
+
+def test_table_group_by():
+    table = pa.table([
+        pa.array([1, 2, 3, 4, 5]),
+        pa.array(["a", "a", "b", "b", "c"]),
+        pa.array([10, 20, 30, 40, 50])
+    ], names=["values", "keys", "bigvalues"])
+
+    r = table.group_by("keys", ["values"], "hash_sum")
+    assert r.to_pydict() == {
+        "key": ["a", "b", "c"],
+        "values_sum": [3, 7, 5]
+    }
```

Comment:
You should probably sort the results so that it's not dependent on underlying implementation details.

Reply:
Uhm, I think I'm missing what you are referring to. I mean, I'm comparing two dicts there, so not sure what should be sorted o_O

Reply:
I mean the list of key/value pairs. The keys and their associated aggregate values may not be output in that order.

Reply:
Isn't that something the test would catch if it ever changes? You would end up with the values for the key into the aggregation result and viceversa if the order of the columns we emit ever changes.

I added a test case where the aggregations are reversed and it still passes, at the moment the function always output the aggregations in the order you provided them.

Reply:
I mean the result is currently:
```python
    {
        "key": ["a", "b", "c"],
        "values_sum": [3, 7, 5]
    }
```

but it could very well be:
```python
    {
        "key": ["a", "c", "b"],
        "values_sum": [3, 5, 7]
    }
```

or any other order

Reply:
oic, you want the test to be more robust in case in the future the order of the keys changes while remaining a still valid result. I'll see what I can do.

Reply:
Ok, sorted the keys in a way that should retain test readability.